src/Evolution/DgSubcell/Actions/TciAndRollback.hpp

```diff
@@ -0,0 +1,252 @@
+        "of a phase.");
+    static_assert(
+        tmpl::count_if<
+            ActionList,
+            std::is_same<tmpl::_1,
+                         tmpl::pin<::Actions::Label<
+                             evolution::dg::subcell::Actions::Labels::
+                                 BeginSubcellAfterDgRollback>>>>::value == 1,
+        "Must have the BeginSubcellAfterDgRollback label exactly once in the "
+        "action list of a phase.");
+
+    using variables_tag = typename Metavariables::system::variables_tag;
+    using dt_variables_tag = db::add_tag_prefix<::Tags::dt, variables_tag>;
+
+    const ActiveGrid active_grid = db::get<Tags::ActiveGrid>(box);
+    ASSERT(active_grid == ActiveGrid::Dg,
+           "Must be using DG when calling TciAndRollback action.");
+
+    const Mesh<Dim>& dg_mesh = db::get<::domain::Tags::Mesh<Dim>>(box);
+    const auto& subcell_mesh = db::get<Tags::Mesh<Dim>>(box);
+
+    // Project candidate solution to subcells
+    db::mutate<Tags::Inactive<variables_tag>>(
+        make_not_null(&box),
+        [&dg_mesh, &subcell_mesh](const auto inactive_vars_ptr,
+                                  const auto& active_vars) noexcept {
+          // Note: strictly speaking, to be conservative this should project uJ
+          // instead of u.
+          fd::project(inactive_vars_ptr, active_vars, dg_mesh,
+                      subcell_mesh.extents());
+        },
+        db::get<variables_tag>(box));
+
+    // Run RDMP TCI since no user info beyond the input file options are needed
+    // for that.
+    std::pair self_id{
+        Direction<Metavariables::volume_dim>::lower_xi(),
+        ElementId<Metavariables::volume_dim>::external_boundary_id()};
+    ASSERT(
+        db::get<Tags::NeighborDataForReconstructionAndRdmpTci<Dim>>(box).count(
+            self_id) != 0,
+        "The self ID is not in the NeighborData.");
+    const NeighborData& self_neighbor_data =
+        db::get<Tags::NeighborDataForReconstructionAndRdmpTci<Dim>>(box).at(
+            self_id);
+    const SubcellOptions& subcell_options = db::get<Tags::SubcellOptions>(box);
+    // Note: we assume the max/min over all neighbors and ourselves at the past
+    // time step has been collected into
+    // `self_neighbor_data.max/min_variables_values`
+    bool cell_is_troubled =
+        rdmp_tci(db::get<variables_tag>(box),
+                 db::get<Tags::Inactive<variables_tag>>(box),
+                 self_neighbor_data.max_variables_values,
+                 self_neighbor_data.min_variables_values,
+                 subcell_options.rdmp_delta0(), subcell_options.rdmp_epsilon());
+
+    // If the RDMP TCI marked the candidate as acceptable, check with the
+    // user-specified TCI, since that could be stricter.
+    if (not cell_is_troubled) {
+      cell_is_troubled = db::mutate_apply<TciMutator>(
+          make_not_null(&box), subcell_options.persson_exponent());
```

Comment:
Ah, good question! I'll add a comment, but the current reasoning is that you should use `exponent+1` for going back to DG, and rather than trying to get that correct via documentation, I figured it's easiest if I always pass the exponent in that people should be using based on what they specified in the input file